README.adoc

```diff
@@ -0,0 +1,65 @@
+= csharp-commons.persistence
+
+== Description
+To be able to query or change data persistently in a database, for example. 
+
+== Class diagramm 
+image:images/Klassendiagramm_commons_persistance.png["Klassendiagramm"]
+
+
+== Example
+We want to use commons.persistance to get the number and a list of all movies contained in a given database. +
+Steps:
+
+* register the Dao
+[source, C#]
+....
+builder.Services.AddScoped<IGenericDao<Movie, int>, MovieDao>();
+....
+
+* create a new class to be able to implement the dao
+[source, C#]
+....
+public class MovieDao : GenericDao<Movie, int>
+    {
+        public MovieDao(MvcMovieContext dbContext) : base(dbContext)
+        {
+        }
+    }
+....
+* instantiate new Dao in the controller class with an integer as Id
+[source, C#]
+....
+private readonly IGenericDao<Movie, int> _movieDao;
+....
+* add the dao to the constructor of the controller
+[source, C#]
+....
+public MoviesController(IGenericDao<Movie, int> movieDao)
```

Comment:
Bibliothek könnte auch in Konsolenanwendung verwendet werden, würde das Beispiel daher noch allgemeiner halten. Vorschlag:
IGenericDao<Movie, int> movieDao = new MovieDao<Movie, int>();
IEnumerable<Movie> allMovies = movieDao.GetAll();